File: gwwz/src/com/control/third/ManageThirdInfo.java
package com.control.third;

import java.io.IOException;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import com.dao.ThirdInfoDAO;
import com.util.PublicToolCheckParam;
import com.util.PublicToolShowThirdInfoList;

public class ManageThirdInfo extends HttpServlet {

	/**
	 * 
	 */
	private static final long serialVersionUID = 3099608914702213565L;
	static ThirdInfoDAO cdao = new ThirdInfoDAO();

	public  static void coolThirdInfo(HttpServletRequest request, HttpServletResponse response)
		throws ServletException, IOException {
			String thirdId = request.getParameter("thirdId");
			String isCooled = request.getParameter("isCooled");
			if(PublicToolCheckParam.checkNullAndEmpty(thirdId) && PublicToolCheckParam.checkNullAndEmpty(isCooled)){
					cdao.coolThird(Integer.parseInt(thirdId),Integer.parseInt(isCooled));
					String isSearching = request.getParameter("isSearching");
					String currentPage = request.getParameter("currentPage");
					if(PublicToolCheckParam.checkNullAndEmpty(isSearching)){
						String key = request.getParameter("key");
						PublicToolShowThirdInfoList.searchThirdInfo(request, response, currentPage, key);
						request.setAttribute("key", key);
						request.setAttribute("isSearching", "yes");
					}
					else{
						PublicToolShowThirdInfoList.showThirdInfoList(request, response, currentPage);
					}
			}
			request.getRequestDispatcher("../third/manageThirdInfo.jsp").forward(request, response);
			
	}

	public static void coolMoreThirdInfo(HttpServletRequest request, HttpServletResponse response)
		throws ServletException, IOException {
		
			String currentPage = request.getParameter("currentPage");
			String[] ckb = request.getParameterValues("ckb");
			if(ckb != null){
				for(int i=0;i<ckb.length;i++){
	//				cdao.coolThird(Integer.parseInt(ckb[i]));
				}
			}
			PublicToolShowThirdInfoList.showThirdInfoList(request, response, currentPage);
			request.getRequestDispatcher("../third/manageThirdInfo.jsp").forward(request, response);
		
	}

	public static void searchThirdInfo(HttpServletRequest request, HttpServletResponse response)
		throws ServletException, IOException {
		
			String currentPage = request.getParameter("currentPage");
			String key = request.getParameter("key");
			PublicToolShowThirdInfoList.searchThirdInfo(request, response, currentPage, key);
			request.setAttribute("isSearching", "yes");
			request.setAttribute("key", key);
			request.getRequestDispatcher("../third/manageThirdInfo.jsp").forward(request, response);
		
	}

	public static void showManageThirdInfoPage(HttpServletRequest request, HttpServletResponse response)
		throws ServletException, IOException {
		
			String currentPage = request.getParameter("currentPage");
			PublicToolShowThirdInfoList.showThirdInfoList(request, response, currentPage);
			request.getRequestDispatcher("../third/manageThirdInfo.jsp").forward(request, response);
	
	}


/**
 * Constructor of the object.
 */
public ManageThirdInfo() {
	super();
}


/**
 * Destruction of the servlet. <br>
 */
public void destroy() {
	super.destroy(); // Just puts "destroy" string in log
	// Put your code here
}

/**
 * The doGet method of the servlet. <br>
 *
 * This method is called when a form has its tag value method equals to get.
 * 
 * @param request the request send by the client to the server
 * @param response the response send by the server to the client
 * @throws ServletException if an error occurred
 * @throws IOException if an error occurred
 */
public void doGet(HttpServletRequest request, HttpServletResponse response)
		throws ServletException, IOException {

	doPost(request, response);
}

/**
 * The doPost method of the servlet. <br>
 *
 * This method is called when a form has its tag value method equals to post.
 * 
 * @param request the request send by the client to the server
 * @param response the response send by the server to the client
 * @throws ServletException if an error occurred
 * @throws IOException if an error occurred
 */
public void doPost(HttpServletRequest request, HttpServletResponse response)
		throws ServletException, IOException {


String method = request.getParameter("method");
if(PublicToolCheckParam.checkNullAndEmpty(method)){
	if(method.equals("showManageThirdInfoPage")){
		showManageThirdInfoPage(request, response);
	}
	else if(method.equals("cool")){
		coolThirdInfo(request, response);
	}
	else if(method.equals("coolMore")){
		coolMoreThirdInfo(request, response);
	}
	else if(method.equals("search")){
		searchThirdInfo(request, response);
	}
}
}

	/**
	 * Initialization of the servlet. <br>
	 *
	 * @throws ServletException if an error occurs
	 */
	public void init() throws ServletException {
		// Put your code here
	}

}

File: gwwz/WebRoot/js/admin/manageThirdInfo.js
function coolMoreThirdInfo(){
	var ckb = document.getElementsByName("ckb");
	var flag = 0;
	var i;
	if(ckb.length == 0){
		alert("当前无任何商户信息!");
		return false;
	}
	else if(ckb.length > 0){
		for(i=0;i<ckb.length;i++){
			if(ckb[i].checked == true){
				flag = 1;
				break;
			}
		}
	}
	if(flag == 1){
		if(confirm("您确定要取消审核当前所选商户?")){
			return true;
		}
		return false;
		
	}
	else{
		alert("请选择您要审核通过的商户!");
		return false;
	}
}


function selectAll(){
	var ckb = document.getElementsByName("ckb");
	if(ckb.length > 0){
		if(document.getElementById("checkAll").checked == true){
			for(var i=0;i<ckb.length;i++){
				ckb[i].checked = true;
			}
		}
		else if(document.getElementById("checkAll").checked == false){
			for(var i=0;i<ckb.length;i++){
				ckb[i].checked = false;
			}
		}
	}
}



function searchThirdInfo(){
	var key = document.getElementById("key").value;
	if(key == ""){
		alert("请输入查询关键词!");
		return false;
	}
	else{
		window.location.href = "../servlet/ManageThirdInfo?method=search&key="+key+"";
		return true;
	}
}

function coolThirdInfoOfSearching(thirdId,isCooled,key){
	if(isCooled == "0"){
		if(confirm("您确定要取消审核当前所选商户?")){
			window.location.href = "../servlet/ManageThirdInfo?method=cool&thirdId="+thirdId+"&isCooled="+isCooled+"&isSearching=yes&key="+key+"";
		}
	}
	else{
		if(confirm("您确定要审核通过当前所选商户?")){
			window.location.href = "../servlet/ManageThirdInfo?method=cool&thirdId="+thirdId+"&isCooled="+isCooled+"&isSearching=yes&key="+key+"";
		}
	}
}

function coolThirdInfo(thirdId,isCooled){
	if(isCooled == "0"){
		if(confirm("您确定要取消审核当前所选商户?")){
			window.location.href = "../servlet/ManageThirdInfo?method=cool&thirdId="+thirdId+"&isCooled="+isCooled+"";
		}
	}
	else{
		if(confirm("您确定要审核当前所选商户?")){
			window.location.href = "../servlet/ManageThirdInfo?method=cool&thirdId="+thirdId+"&isCooled="+isCooled+"";
		}
	}
}




function back(){

	window.location.href = "../servlet/ManageThirdInfo?method=showManageThirdInfoPage";
	}
	
	